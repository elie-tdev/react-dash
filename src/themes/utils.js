export const opacity = (hex, opacity) => {
  var result = /^#?([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})$/i.exec(hex);
  if (!result) {
    throw new Error('Invalid hex', hex);
  }
  return `rgba(${parseInt(result[1], 16)}, ${parseInt(result[2], 16)}, ${parseInt(
    result[3],
    16,
  )}, ${opacity})`;
};

export const gradient = ({ type = 'linear', colors, background }) =>
  `${type}-gradient(${colors.join(', ')}), ${background};`;
