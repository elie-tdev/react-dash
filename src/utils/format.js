const formatter = new Intl.NumberFormat('en-US', {
  style: 'currency',
  currency: 'USD',
  minimumFractionDigits: 2,
});

const parseFloat = (str, val = 2) => {
  str = str && str.toString();
  str = str && str.indexOf('.') > -1 ? str.slice(0, str.indexOf('.') + val + 1) : str;
  return Number(str);
};

export const formatDollar = number => formatter.format(parseFloat(number, 2));
