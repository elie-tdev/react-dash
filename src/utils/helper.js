const maskAccountNumber = accNumber => {
  return accNumber && <>&bull;&bull;&bull;&bull; {accNumber.slice(accNumber.length - 4)}</>;
};

export default maskAccountNumber;
