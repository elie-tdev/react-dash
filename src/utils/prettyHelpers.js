export const prettyCurrency = (amount, currency = 'USD') => {
  if (amount == null) {
    return false;
  }
  if (amount < 0) {
    amount = 0;
  }
  return amount.toLocaleString('en', { style: 'currency', currency });
};

export const prettyDateTime = time => {
  if (time == null) {
    return false;
  }
  let d = new Date(time);
  return (
    d.toLocaleDateString('en', {
      month: '2-digit',
      day: '2-digit',
      year: '2-digit',
    }) +
    ' at ' +
    d.toLocaleTimeString('en', { hour: '2-digit', minute: '2-digit' }).replace(/^0(?:0:0?)?/, '')
  );
};

export const prettyDate = time => new Date(time).toLocaleDateString('en', {});

export const prettyTime = time =>
  new Date(time).toLocaleTimeString([], { hour: '2-digit', minute: '2-digit' });

export const prettyHour = time =>
  new Date(time)
    .toLocaleTimeString('en', { hour: '2-digit', minute: '2-digit' })
    .replace(/^0(?:0:0?)?/, '');

export const prettyPhone = number =>
  number.substr(2).replace(/(\d{1,2})(\d{1})?(\d{1,3})?(\d{1,4})?/, function (_, p1, p2, p3, p4) {
    let output = '';
    if (p1) output = `${p1}`;
    if (p2) output += `${p2}-`;
    if (p3) output += `${p3}-`;
    if (p4) output += `${p4}`;
    return output;
  });

export const minimumDays = (date, numDays) => {
  const d = new Date(date);
  const totalNumDays = daysIntoYear(d) + numDays;
  return Boolean(daysIntoYear(d) >= totalNumDays || totalNumDays <= 365);
};

export const daysIntoYear = date => {
  return (
    (Date.UTC(date.getFullYear(), date.getMonth(), date.getDate()) -
      Date.UTC(date.getFullYear(), 0, 0)) /
    24 /
    60 /
    60 /
    1000
  );
};

export const cleanAmountRegex = value => {
  return value.replace(/[^0-9]/g, '');
};

export const cleanAmount = value => {
  const pos = value.indexOf('.');
  let result;
  if (pos !== -1) {
    const part1 = value.substr(0, pos);
    const part2 = value.substr(pos + 1);
    result = cleanAmountRegex(part1) + '.' + cleanAmountRegex(part2);
  } else {
    result = cleanAmountRegex(value);
  }
  return result;
};
