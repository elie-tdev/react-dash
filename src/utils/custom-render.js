import React from 'react';
import { I18nextProvider } from 'react-i18next';
import { render } from '@testing-library/react';
import { CssBaseline } from '@material-ui/core';
import { BrowserRouter as Router } from 'react-router-dom';
import { ThemeProvider } from '@material-ui/core/styles';
import { MockedProvider } from '@apollo/client/testing';

import { MenuContextProvider } from '@/contexts';
import i18n from '@/translations';
import { LightTheme } from '@/themes';

const wrapper = ({ mocks = [], menuContextProviderProps = {} }) => {
  const Wrapper = ({ children }) => {
    return (
      <Router>
        <MockedProvider mocks={mocks} addTypename={false}>
          <ThemeProvider theme={LightTheme}>
            <MenuContextProvider {...menuContextProviderProps}>
              <CssBaseline />
              <I18nextProvider i18n={i18n}>{children}</I18nextProvider>
            </MenuContextProvider>
          </ThemeProvider>
        </MockedProvider>
      </Router>
    );
  };

  return Wrapper;
};

const customRender = (ui, { mocks, menuContextProviderProps, ...restOptions } = {}) =>
  render(ui, { wrapper: wrapper({ mocks, menuContextProviderProps }), ...restOptions });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
