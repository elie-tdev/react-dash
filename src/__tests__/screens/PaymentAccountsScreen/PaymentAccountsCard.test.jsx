import React from 'react';

import PaymentAccountsCard from '@/screens/PaymentAccountsScreen/PaymentAccountsCard';
import { render } from '@/utils/custom-render';

describe('PaymentAccountsCard', () => {
  test('renders PaymentAccountsCard', () => {
    const { container } = render(<PaymentAccountsCard />);
    expect(container).toMatchSnapshot();
  });
});
