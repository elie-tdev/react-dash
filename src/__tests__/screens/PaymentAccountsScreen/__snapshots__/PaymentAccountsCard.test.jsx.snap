// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaymentAccountsCard renders PaymentAccountsCard 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
  >
    <div
      class="MuiBox-root MuiBox-root-52 makeStyles-row-20 makeStyles-header-42"
    >
      <div
        class="MuiBox-root MuiBox-root-53 makeStyles-arrowBackIconContainer-46"
      >
        <div
          class="MuiBox-root MuiBox-root-54"
        >
          <button
            class="MuiButtonBase-root MuiIconButton-root makeStyles-arrowBackIconButton-45"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root makeStyles-backIcon-44"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <h4
            class="MuiTypography-root makeStyles-pageTitle-51 MuiTypography-h4"
          >
            Payment Accounts
          </h4>
        </div>
      </div>
      <div
        class="MuiBox-root MuiBox-root-55 makeStyles-addIconContainer-49"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root makeStyles-addIcon-50"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
  <div
    class="MuiGrid-root makeStyles-nonMobileGridItem-9 MuiGrid-item MuiGrid-grid-xs-12"
  >
    <div
      class="MuiBox-root MuiBox-root-59 makeStyles-paymentAccountsBox-5 makeStyles-paymentAccountsBox-57"
    >
      <div
        class="MuiBox-root MuiBox-root-60 makeStyles-lockIconBox-7 makeStyles-lockIconBox-58"
      >
        <div
          class="MuiBox-root MuiBox-root-61 makeStyles-lockIconContainer-8"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M21 18v1c0 1.1-.9 2-2 2H5c-1.11 0-2-.9-2-2V5c0-1.1.89-2 2-2h14c1.1 0 2 .9 2 2v1h-9c-1.11 0-2 .9-2 2v8c0 1.1.89 2 2 2h9zm-9-2h10V8H12v8zm4-2.5c-.83 0-1.5-.67-1.5-1.5s.67-1.5 1.5-1.5 1.5.67 1.5 1.5-.67 1.5-1.5 1.5z"
            />
          </svg>
        </div>
      </div>
      <h1
        class="MuiTypography-root makeStyles-row-6 MuiTypography-h1"
      >
        Add a Payment Account
      </h1>
      <p
        class="MuiTypography-root makeStyles-caption-4 makeStyles-row-6 MuiTypography-body1"
      >
        Your default bank account will be used for any future payments scheduled through the dashboard. It will not impact previously scheduled payments.
      </p>
      <button
        class="MuiButtonBase-root MuiButton-root makeStyles-root-62 makeStyles-root-63 MuiButton-contained MuiButton-containedSizeLarge MuiButton-sizeLarge MuiButton-disableElevation"
        data-testid="PrimaryButton"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          + New Payment Account
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
